pkg/migration/migrations/protected_ts_meta_migration.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package migrations
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/migration"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkeys"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/systemschema"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+)
+
+func protectedTsMetaPrivilegesMigration(
+	ctx context.Context, _ clusterversion.ClusterVersion, c migration.Cluster,
+) error {
+	return c.DB().Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+		log.Infof(ctx, "%s", "updating privileges in system.protected_ts_meta descriptor")
```

Comment:
do you have interest in using `descs.Txn`? I think you should have all of the dependencies if you declare it as a SQL migration. There is an important distinction: we run the SQL migrations on serverless tenants.

Reply:
I'd rather not use `descs.Txn` here for various reasons:
1. I want to be sure I'm reading the descriptor from storage and not from a system descriptor cache.
2. I want to avoid using the `catalogkv` API because it runs post-deserialization changes which fix the descriptors it reads, I could use it but I feel that the change we're doing is a bit more low-level than that (we're not changing version numbers, etc.).

Reply:
> storage and not from a system descriptor cache.

I'm scared that this is a concern for mutable descriptor resolution.

> but I feel that the change we're doing is a bit more low-level than that (we're not changing version numbers, etc.).

We could/should change version numbers, no?

Reply:
I was (probably mistakenly) assuming that we wouldn't want to change the version numbers, but if we do, then we should also bump them in `systemschema.ProtectedTimestampsMetaTable`, right?

Reply:
I don't know about that. I'm reasonably certain there have been a number of migrations in the past which certainly did not update the in-memory version number. 

Let me ask a different question: what do those in-memory descriptors actually get used for?